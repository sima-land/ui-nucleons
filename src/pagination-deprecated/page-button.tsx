import { cx } from './utils';

export interface PageButtonProps
  extends Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'type'> {
  rounded?: 'all' | 'none' | 'left' | 'right';
  className?: string;
  selected?: boolean;
}

/**
 * Компонент кнопки страницы по умолчанию. Поддерживаются свойства элемента button.
 * @param props Свойства.
 * @param props.selected Выбрана ли кнопка.
 * @param props.className CSS-класс.
 * @param props.rounded С какой стороны делать скругления.
 * @return Элемент.
 * @deprecated
 */
export const PageButton: React.FC<PageButtonProps> = ({
  selected,
  rounded = 'all',
  className,
  ...restProps
}) => (
  <button
    {...restProps}
    type='button'
    className={cx(
      'page-button',
      selected && 'selected',
      rounded === 'all' && 'rounds-all',
      rounded === 'left' && 'rounds-left',
      rounded === 'right' && 'rounds-right',
      className,
    )}
  />
);
