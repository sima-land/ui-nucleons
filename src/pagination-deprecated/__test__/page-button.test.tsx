import React from 'react';
import { mount } from 'enzyme';
import { PageButton } from '../page-button';

describe('<PageButton />', () => {
  it('should render without props', () => {
    const wrapper = mount(<PageButton />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle "selected" prop', () => {
    const wrapper = mount(<PageButton selected />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ selected: false });
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle regular button props', () => {
    const spy = jest.fn();
    const wrapper = mount(<PageButton disabled onClick={spy} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle "rounded" prop', () => {
    const wrapper = mount(<PageButton />);

    expect(wrapper.find('button').hasClass('rounds-all')).toBe(true);
    expect(wrapper.find('button').hasClass('rounds-left')).toBe(false);
    expect(wrapper.find('button').hasClass('rounds-right')).toBe(false);

    wrapper.setProps({ rounded: 'left' });
    expect(wrapper.find('button').hasClass('rounds-all')).toBe(false);
    expect(wrapper.find('button').hasClass('rounds-left')).toBe(true);
    expect(wrapper.find('button').hasClass('rounds-right')).toBe(false);

    wrapper.setProps({ rounded: 'right' });
    expect(wrapper.find('button').hasClass('rounds-all')).toBe(false);
    expect(wrapper.find('button').hasClass('rounds-left')).toBe(false);
    expect(wrapper.find('button').hasClass('rounds-right')).toBe(true);

    wrapper.setProps({ rounded: 'none' });
    expect(wrapper.find('button').hasClass('rounds-all')).toBe(false);
    expect(wrapper.find('button').hasClass('rounds-left')).toBe(false);
    expect(wrapper.find('button').hasClass('rounds-right')).toBe(false);
  });
});
