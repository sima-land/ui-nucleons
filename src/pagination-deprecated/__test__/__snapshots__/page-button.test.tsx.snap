// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageButton /> should handle "selected" prop 1`] = `
<PageButton
  selected={true}
>
  <button
    className="page-button selected rounds-all"
    type="button"
  />
</PageButton>
`;

exports[`<PageButton /> should handle "selected" prop 2`] = `
<PageButton
  selected={false}
>
  <button
    className="page-button rounds-all"
    type="button"
  />
</PageButton>
`;

exports[`<PageButton /> should handle regular button props 1`] = `
<PageButton
  disabled={true}
  onClick={[MockFunction]}
>
  <button
    className="page-button rounds-all"
    disabled={true}
    onClick={[MockFunction]}
    type="button"
  />
</PageButton>
`;

exports[`<PageButton /> should render without props 1`] = `
<PageButton>
  <button
    className="page-button rounds-all"
    type="button"
  />
</PageButton>
`;
